src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -225,24 +216,83 @@ private int LookupAggregatorStore(string[] tagKeys, object[] tagValues, int leng
                             return -1;
                         }
 
-                        // Note: We are using storage from ThreadStatic, so need to make a deep copy for Dictionary storage.
-                        if (seqKey == null)
+                        lock (this.tagsToMetricPointIndexDictionary)
```

Comment:
Use a dedicated lock object instead of the default one from the dict (since the dict's own implementation might use it for its internal operation).

Reply:
Curious - would `SpinWait` give better behavior under contention (less CSWitch)?

Reply:
Would be good to explore as follow ups.

Reply:
I have created the issue #2843 to track this.